### PR TITLE
Fix #9 and Large Refactor

### LINKED ISSUE #9: 'Sort' order of Daily Notes entry
In 0.0.3, I just noticed that when I add Obsidian Memos, new entries go to the bottom of the list under the defined heading, but your clips go to the top of the list, so they're out of order.  Any chance to put them at the bottom, or provide a configuration for which you prefer?  Thanks!

### PR DESCRIPTION
**Pull request by jgchristopher**
- #9 support. I had to not allow '#' Level 1 headers in clipped content but appending and prepending now works
- Large Refactor of the codebase to make it easier to move forward
- Use Svelte for settings configuration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This plugin allows you to capture highlights from the web as Markdown entries in
 
 ## Requirements
 
-Currently, Obsidian Clipper only writes to the Daily Note, but I have plans to allow other locations.
+Obsidian Clipper can be configured to write to either the a daily note, weekly note, or both. I have plans to allow other locations.
```
Comment (jgchristopher) [inline README.md:7]: Rewrite this sentence.

File: src/main.ts [hunk 2]
```diff
@@ -1,502 +1,119 @@
...
+    this.plugin = plugin;
+    init(this.plugin);
+  }
+
+  display(): void {
+    const { containerEl } = this;
+
+    containerEl.empty();
+
+    //const { subscribe, set, update } = writable(this.plugin.settings);
```
Comment (jgchristopher) [inline src/main.ts:104]: remove commented out code

File: src/periodicnotes/filewriter.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    const cache = this.app.metadataCache.getFileCache(file);
+    if (cache) {
+      try {
+        let cachedHeadings = cache.headings;
+        if (cachedHeadings) {
+          // We need to see if the configured heading exists in the document
+          const foundHeadingIndex = cachedHeadings.findIndex(
+            (cachedHeading) => {
+              if (
+                cachedHeading.heading === heading &&
```
Comment (jgchristopher) [inline src/periodicnotes/filewriter.ts:149]: Just return this and get rid of the if/else

File: src/settings/BookmarkletSettingsGroup.svelte [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<script lang="ts">
+	import { BookmarketlGenerator } from "src/bookmarkletgenerator";
```
Comment (jgchristopher) [inline src/settings/BookmarkletSettingsGroup.svelte:2]: format

File: src/periodicnotes/periodicnoteentry.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    this.handleWrite(
+      note.path,
+      await noteEntry.formattedEntry(),
+      this.sectionPosition,
+      heading
+    );
+  }
+
+  protected async getNote() {
+    let now = window.moment();
```
Comment (nicknisi) [inline src/periodicnotes/periodicnoteentry.ts:49]:
```suggestion
    const now = window.moment();
```

Comment (nicknisi) [inline src/periodicnotes/periodicnoteentry.ts:49]: **question:** Is there a reason you're getting at `moment` from `window`? Could it be imported instead?

Comment (nicknisi) [reply src/periodicnotes/periodicnoteentry.ts:49]: as a follow-up, if `window` is the proper place, then maybe replace it with `globalThis`

```suggestion
    const now = globalThis.moment();
```

File: src/periodicnotes/sectionposition.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const SectionPosition = {
+	PREPEND: "prepend",
+	APPEND: "append",
+};
+export type SectionPosition =
+	typeof SectionPosition[keyof typeof SectionPosition];
```
Comment (nicknisi) [inline src/periodicnotes/sectionposition.ts:6]: **issue:** The type here is just `string` when I think you mean for it to be `'prepend' | 'append'`.

```suggestion
export type SectionPosition = 'prepend' | 'append';
```

Comment (nicknisi) [reply src/periodicnotes/sectionposition.ts:6]: Another suggestion: you might be able to get away without the object literal and just relying on string comparison. With this type, you'll have the same amount of type safety.

File: src/periodicnotes/weeklyperiodicnoteentry.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  protected getPeriodicNote(moment: Moment, allNotes: Record<string, TFile>) {
+    return getWeeklyNote(moment, allNotes);
+  }
+
+  protected hasPeriodicNoteEnabled() {
+    return appHasWeeklyNotesPluginLoaded();
+  }
+
+  protected async waitForNoteCreation(moment: Moment) {
+    let weeklyNote = await createWeeklyNote(moment);
```
Comment (nicknisi) [inline src/periodicnotes/weeklyperiodicnoteentry.ts:32]:
```suggestion
    const weeklyNote = await createWeeklyNote(moment);
```

File: src/settings/CommonSettingsGroup.svelte [hunk 8]
```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+	import { settings } from "./settingsstore";
+</script>
+
+<div class="clp_section_margin">
```
Comment (nicknisi) [inline src/settings/CommonSettingsGroup.svelte:5]: **nitpick:** This file looks like it's tabs and others are spaces. Maybe set up editorconfig or prettier.

File: src/settings/settingsstore.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import type ObsidianClipperPlugin from "src/main";
+import type { ObsidianClipperSettings } from "src/settings";
+import { type Writable, writable } from "svelte/store";
```
Comment (nicknisi) [inline src/settings/settingsstore.ts:3]: **praise:** whoa, til about `import { type ... }` 😱

File: tsconfig.json [hunk 10]
```diff
@@ -10,13 +10,14 @@
     "moduleResolution": "node",
     "importHelpers": true,
     "isolatedModules": true,
-	"strictNullChecks": true,
+	  "strictNullChecks": true,
```
Comment (nicknisi) [inline tsconfig.json:13]: **nitpick:** stray tab
```suggestion
      "strictNullChecks": true,
```

**Review by nicknisi — CHANGES_REQUESTED** [review]
Looks great overall! It might be worthwhile to setup Prettier and eslint to help prevent most of the nitpicks I had!

File: src/periodicnotes/sectionposition.ts (hunk 6 above)
Comment (jgchristopher) [reply src/periodicnotes/sectionposition.ts]: I was missing an `as const` after the SectionPosition object literal. Adding that gave the intended behavior.

File: src/periodicnotes/filewriter.ts [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+    heading?: string
+  ): Promise<TFile> {
+    if (!heading) {
+      let fileData = await this.app.vault.read(file);
+      return this.writeAndOpenFile(
+        file.path,
+        this.positionDataWithNoHeader(fileData, clippedData)
+      );
+    } else {
+      let insertSection: { firstLine: number; lastLine?: number } = {
```
[the remark this answers is not in the document]
Comment (jgchristopher) [reply src/periodicnotes/filewriter.ts:47]: This is working but I need to look at this code more and simplify

**Pull request merged**